src/main/java/com/dj/core/model/graph/ConnectedNeuron.java

```diff
@@ -190,8 +187,20 @@ public void addForwardConnection(final Neuron neuron) {
 
     @Override
     public void addBackwardConnection(final Neuron neuron, final Double weight) {
-        backwardConnections.put(neuron, weight);
-        inputSignals.put(neuron, Double.NaN);
+        int nextIndex = neuronIndexes.size();
+        neuronIndexes.put(neuron, nextIndex);
+
+        backwardConnections = addToTensor(backwardConnections, nextIndex, weight);
+        inputSignals = addToTensor(inputSignals, nextIndex, Double.NaN);
+    }
+
+    private RealMatrix addToTensor(RealMatrix tensor, int newColumnIndex, double value) {
```

Comment:
please make input argument final, just to be consistent with other places (or remove final in other places :) )

Reply:
Done